buffother.lic

```diff
@@ -0,0 +1,47 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#buffother
+=end
+custom_require.call(%w(common common-arcana events spellmonitor drinfomon common-travel))
+class Buffother
+  include DRC
+  include DRCA
+  include DRCT
+
+  def initialize
+    arg_definitions = [
+      [
+        { name: 'recipient', regex: /\w+/, description: 'Person to buff' },
+        { name: 'waggle', regex: /\w+/, description: 'Waggle set to cast' },
+        { name: 'room', regex: /\d+/, optional: true, description: 'Room to move to' }
+      ]
+    ]
+    args = parse_args(arg_definitions)
+    @settings = get_settings
+    @recipient = args.recipient.capitalize
+    @waggle = args.waggle
+    @room = args.room
+    move_to_room
+  end
+
+  def move_to_room
+    echo "Moving to room"
+    DRCT.walk_to(@room)
+    cast_buff_spells
+    pause 1
+  end
+
+  def cast_buff_spells
+    echo "In cast_buff_spells"
+    data = @settings.waggle_sets["#{@waggle}"].values.each do |data|
+      data['cast'] = "cast #{@recipient}"
+      DRCA.do_buffs(@settings, @waggle)
```

Comment:
Note, because of https://github.com/rpherbig/dr-scripts/blob/master/common-arcana.lic#L355, if this character who is casting the spells somehow already has the same spell active on themself, I don't think the buff will end up being cast on _the other character_.

Reply:
They'll need to set the recast: value high in the waggle they're calling. i.e. recast: 100 instead of recast: 1 or recast: 2 which is more normal. As long as the buff on themself has a duration of less than that large value, it'll be fine.